Index: suricata-7.0.10/src/flow.c
===================================================================
--- suricata-7.0.10.orig/src/flow.c
+++ suricata-7.0.10/src/flow.c
@@ -469,6 +469,13 @@ void FlowHandlePacketUpdate(Flow *f, Pac
         } else if (PKT_IS_IPV6(p)) {
             FlowUpdateTtlTS(f, p, IPV6_GET_HLIM(p));
         }
+#ifdef NFQ
+    if (!f->hw_address_src_known && p->nfq_v.hw_address_known){
+        SCLogDebug("setting src MAC address %02x:%02x:%02x:%02x:%02x:%02x", p->nfq_v.hw_address[0], p->nfq_v.hw_address[1], p->nfq_v.hw_address[2], p->nfq_v.hw_address[3], p->nfq_v.hw_address[4], p->nfq_v.hw_address[5]);
+        f->hw_address_src_known = 1;
+        memcpy(f->hw_address_src, p->nfq_v.hw_address, 6);
+    }
+#endif
     } else {
         f->tosrcpktcnt++;
         f->tosrcbytecnt += GET_PKT_LEN(p);
@@ -491,6 +498,13 @@ void FlowHandlePacketUpdate(Flow *f, Pac
         } else if (PKT_IS_IPV6(p)) {
             FlowUpdateTtlTC(f, p, IPV6_GET_HLIM(p));
         }
+#ifdef NFQ
+    if (!f->hw_address_dst_known && p->nfq_v.hw_address_known){
+        SCLogDebug("setting dst MAC address %02x:%02x:%02x:%02x:%02x:%02x", p->nfq_v.hw_address[0], p->nfq_v.hw_address[1], p->nfq_v.hw_address[2], p->nfq_v.hw_address[3], p->nfq_v.hw_address[4], p->nfq_v.hw_address[5]);
+        f->hw_address_dst_known = 1;
+        memcpy(f->hw_address_dst, p->nfq_v.hw_address, 6);
+    }
+#endif
     }
 
     if (f->flow_state == FLOW_STATE_ESTABLISHED) {
Index: suricata-7.0.10/src/flow.h
===================================================================
--- suricata-7.0.10.orig/src/flow.h
+++ suricata-7.0.10/src/flow.h
@@ -490,6 +490,12 @@ typedef struct Flow_
     uint32_t tosrcpktcnt;
     uint64_t todstbytecnt;
     uint64_t tosrcbytecnt;
+#ifdef NFQ
+    uint8_t hw_address_src[6];
+    uint8_t hw_address_src_known;
+    uint8_t hw_address_dst[6];
+    uint8_t hw_address_dst_known;
+#endif
 } Flow;
 
 enum FlowState {
Index: suricata-7.0.10/src/source-nfq.c
===================================================================
--- suricata-7.0.10.orig/src/source-nfq.c
+++ suricata-7.0.10/src/source-nfq.c
@@ -405,6 +405,7 @@ static int NFQSetupPkt (Packet *p, struc
     int ret;
     char *pktdata;
     struct nfqnl_msg_packet_hdr *ph;
+    struct nfqnl_msg_packet_hw *hwaddr;
 
     // Early release function -- will be updated once repeat
     // mode handling has been done
@@ -415,6 +416,13 @@ static int NFQSetupPkt (Packet *p, struc
         p->nfq_v.id = SCNtohl(ph->packet_id);
         p->nfq_v.hw_protocol = ph->hw_protocol;
     }
+    hwaddr = nfq_get_packet_hw(tb);
+    if (hwaddr != NULL) {
+        p->nfq_v.hw_address_known = 1;
+        memcpy(p->nfq_v.hw_address, hwaddr->hw_addr, 6);
+    } else {
+        p->nfq_v.hw_address_known = 0;
+    }
     /* coverity[missing_lock] */
     p->nfq_v.mark = nfq_get_nfmark(tb);
     if (nfq_config.mode == NFQ_REPEAT_MODE) {
Index: suricata-7.0.10/src/source-nfq.h
===================================================================
--- suricata-7.0.10.orig/src/source-nfq.h
+++ suricata-7.0.10/src/source-nfq.h
@@ -46,6 +46,8 @@ typedef struct NFQPacketVars_
     uint32_t ifi;
     uint32_t ifo;
     uint16_t hw_protocol;
+    uint8_t hw_address_known;
+    uint8_t hw_address[6];
 } NFQPacketVars;
 
 typedef struct NFQQueueVars_
Index: suricata-7.0.10/src/flow-util.h
===================================================================
--- suricata-7.0.10.orig/src/flow-util.h
+++ suricata-7.0.10/src/flow-util.h
@@ -27,13 +27,24 @@
 #include "flow.h"
 #include "stream-tcp-private.h"
 
+#ifdef NFQ
 #define RESET_COUNTERS(f)                                                                          \
     do {                                                                                           \
         (f)->todstpktcnt = 0;                                                                      \
         (f)->tosrcpktcnt = 0;                                                                      \
         (f)->todstbytecnt = 0;                                                                     \
         (f)->tosrcbytecnt = 0;                                                                     \
+	(f)->hw_address_src_known = 0;                                                             \
+	(f)->hw_address_dst_known = 0;                                                             \
     } while (0)
+#else
+#define RESET_COUNTERS(f) do { \
+        (f)->todstpktcnt = 0; \
+        (f)->tosrcpktcnt = 0; \
+        (f)->todstbytecnt = 0; \
+        (f)->tosrcbytecnt = 0; \
+    } while (0)
+#endif
 
 #define FLOW_INITIALIZE(f)                                                                         \
     do {                                                                                           \
