Index: suricata-8.0.0/rust/Makefile.am
===================================================================
--- suricata-8.0.0.orig/rust/Makefile.am
+++ suricata-8.0.0/rust/Makefile.am
@@ -64,10 +64,12 @@ CARGO_ENV +=	LOCALSTATEDIR=$(e_localstat
 # To pass specific CFLAGS to Lua.
 CARGO_ENV +=	SURICATA_LUA_SYS_CFLAGS="$(SURICATA_LUA_SYS_CFLAGS)"
 
+CARGO_ENV +=	RUSTFLAGS="-C target-feature=-crt-static"
+
 all-local: Cargo.toml
 	mkdir -p $(abs_top_builddir)/rust/gen
 	cd $(abs_top_srcdir)/rust && \
-		CC="$(CC)" AR="$(AR)" RANLIB="$(RANLIB)"\
+		CC="$(CC)" AR="$(AR)" RANLIB="$(RANLIB)" LD="$(CC)" \
 		$(CARGO_ENV) \
 		$(CARGO) build $(RELEASE) $(NIGHTLY_ARGS) \
 			--features "$(RUST_FEATURES)" $(RUST_TARGET)
Index: suricata-8.0.0/rust/.cargo/config.toml.in
===================================================================
--- suricata-8.0.0.orig/rust/.cargo/config.toml.in
+++ suricata-8.0.0/rust/.cargo/config.toml.in
@@ -6,3 +6,5 @@
 @rust_vendor_comment@
 @rust_vendor_comment@[source.vendored-sources]
 @rust_vendor_comment@directory = '@e_rustdir@/vendor'
+@rust_vendor_comment@[target.@host@]
+@rust_vendor_comment@linker = "@target_alias@-gcc"
