From e04755a2af383d449e8d502e7fb28193c4e62e6b Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Fri, 22 Aug 2025 23:52:51 +0200
Subject: [PATCH 2/2] Drop currently unimplemented pages

---
 ui/pages/dashboard.slint | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/ui/pages/dashboard.slint b/ui/pages/dashboard.slint
index c114fc9..5bbd2ff 100644
--- a/ui/pages/dashboard.slint
+++ b/ui/pages/dashboard.slint
@@ -54,10 +54,7 @@ export component Dashboard inherits Page {
 
     // Handle navigation keys
     function navigate(key: string) {
-        if (key == Key.RightArrow && State.current-page == Page.dashboard) {
-            debug("Getting menu");
-            State.current-page = Page.menu;
-        } else if (key == Key.RightArrow && State.current-page == Page.qr-details) {
+        if (key == Key.RightArrow && State.current-page == Page.qr-details) {
             debug("Getting QR");
             State.current-page = Page.qr;
         } else if (key == Key.LeftArrow && State.current-page == Page.qr) {
@@ -72,12 +69,6 @@ export component Dashboard inherits Page {
         } else if (key == Key.DownArrow && (State.current-page == Page.qr || State.current-page == Page.qr-details)) {
             debug("Getting Dashboard");
             State.current-page = Page.dashboard;
-        } else if (key == Key.DownArrow && State.current-page == Page.dashboard) {
-            debug("Getting loading spinner");
-            State.current-page = Page.loading;
-        } else if (key == Key.UpArrow && State.current-page == Page.loading) {
-            debug("Getting Dashboard");
-            State.current-page = Page.dashboard;
         } else {
             debug("Unhandled key on Dashboard page: " + key);
         }
-- 
2.51.0

