From fe72a46929ffc3d2a86bb4d6e29d400c759bfccd Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Mon, 1 Sep 2025 07:09:45 +0200
Subject: [PATCH 4/6] dashboard: Nicer bandwidth display

---
 ui/pages/dashboard.slint | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/ui/pages/dashboard.slint b/ui/pages/dashboard.slint
index 85c7943..2574e3b 100644
--- a/ui/pages/dashboard.slint
+++ b/ui/pages/dashboard.slint
@@ -140,11 +140,21 @@ export component Dashboard inherits Page {
             spacing: 16px;
             alignment: start;
 
+            pure function formatted_bandwidth(bw: int) -> string {
+                if (bw > 1024 * 1024 * 1024 * 10) {
+                    return "\{Math.floor(bw / (1024 * 1024 * 1024))} GiB/s";
+                } else if (bw > 1024 * 1024 * 10) {
+                    return "\{Math.floor(bw / (1024 * 1024))} MiB/s";
+                } else {
+                    return "\{Math.floor(bw / 1024)} KiB/s";
+                }
+            }
+
             HorizontalLayout {
                 spacing: 4px;
 
                 Text {
-                    text: "\{(Readouts.wan-bandwidth.rx / 10 / 1024).floor()} Kib/s"; // Dynamic download speed
+                    text: formatted_bandwidth(Readouts.wan-bandwidth.rx / 10); // Dynamic download speed
                     color: Theme.foreground;
                     font-size: Theme.font-size-normal;
                     font-weight: Theme.font-weight-bold;
@@ -164,7 +174,7 @@ export component Dashboard inherits Page {
                 spacing: 4px;
 
                 Text {
-                    text: "\{(Readouts.wan-bandwidth.tx / 10 / 1024).floor()} Kib/s"; // Dynamic upload speed
+                    text: formatted_bandwidth(Readouts.wan-bandwidth.tx / 10); // Dynamic upload speed
                     color: Theme.foreground;
                     font-size: Theme.font-size-normal;
                     font-weight: Theme.font-weight-bold;
-- 
2.51.0

