From af9ce318070f1d4f239a6ecfe525c4f2abd9e1cf Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Mon, 8 Sep 2025 16:27:19 +0200
Subject: [PATCH 6/6] qr: Display only when Wi-Fi has a password set

---
 ui/pages/dashboard.slint | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/pages/dashboard.slint b/ui/pages/dashboard.slint
index 86ae3ff..a150be3 100644
--- a/ui/pages/dashboard.slint
+++ b/ui/pages/dashboard.slint
@@ -63,7 +63,7 @@ export component Dashboard inherits Page {
         } else if (key == Key.LeftArrow) {
             debug("Getting notifications");
             State.current-page = Page.notifications;
-        } else if (key == Key.UpArrow && State.current-page == Page.dashboard) {
+        } else if (key == Key.UpArrow && State.current-page == Page.dashboard && !(Readouts.initial-setup-finished ? Readouts.ap-guest : Readouts.ap-primary).psk.is-empty) {
             debug("Getting QR");
             State.current-page = Page.qr;
         } else if (key == Key.DownArrow && (State.current-page == Page.qr || State.current-page == Page.qr-details)) {
-- 
2.51.0

