--- a/Cargo.toml	2025-08-22 08:16:56.219916961 +0200
+++ b/Cargo.toml	2025-08-22 08:17:42.112295299 +0200
@@ -35,7 +35,7 @@
 network-interface    = "2"
 qrcode               = "0.14"
 regex                = "1"
-rustix               = { version = "1"   , features = ["fs", "system"] }
+rustix               = { version = "1"   , features = ["fs", "system", "net"] }
 serde                = { version = "1"   , features = ["derive"] }
 serde_json           = "1"
 time                 = { version = "0.3" , features = ["macros"] }
