From f0cb837887b71dff4164684e539506f6c6a5626e Mon Sep 17 00:00:00 2001
From: Lukas Jelinek <lukas.jelinek@nic.cz>
Date: Tue, 3 Oct 2023 12:07:38 +0200
Subject: [PATCH] scripts/syslog-ng.sh: Disable logging to /srv by default

---
 scripts/syslog-ng.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/syslog-ng.sh b/scripts/syslog-ng.sh
index 3ee989d..5fe5873 100755
--- a/scripts/syslog-ng.sh
+++ b/scripts/syslog-ng.sh
@@ -3,7 +3,7 @@
 
 config_load "storage"
 config_get uuid "srv" "uuid"
-config_get_bool syslog "srv" "syslog" "1"
+config_get_bool syslog "srv" "syslog" "0"
 
 # Ignore if storage is not configured
 [ -n "$uuid" ] || return 0
-- 
2.42.0

