--- a/rainbow/led_activity.sh
+++ b/rainbow/led_activity.sh
@@ -85,8 +85,10 @@ supported_activities() {
 				| sort -u
 		done
 	else
-		tr ' ' '\n' <"$(led2sysfs "$led")/trigger" \
-			| grep "$@" || true
+		if [ -e "$(led2sysfs "$led")/trigger" ]; then
+			tr ' ' '\n' <"$(led2sysfs "$led")/trigger" \
+				| grep "$@" || true
+		fi
 	fi
 }
 
@@ -135,6 +137,7 @@ apply_activity() {
 	local trigger="${1:-}"
 	local sysfs
 	sysfs="$(led2sysfs "$led")"
+	[ -e "$sysfs/trigger" ] || return
 	case "$trigger" in
 		activity|activity-inverted|heartbeat|heartbeat-inverted)
 			local systrigger="${trigger%-inverted}"
