From b71e60a61f9857fa01e29eda1ad2227a3a460117 Mon Sep 17 00:00:00 2001
From: Stepan Henek <stepan.henek@nic.cz>
Date: Thu, 20 Nov 2025 10:22:49 +0100
Subject: [PATCH] Remove LuCI link when device is shield

---
 js/src/main/pages.js | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/js/src/main/pages.js b/js/src/main/pages.js
index a97d0089..e6009709 100644
--- a/js/src/main/pages.js
+++ b/js/src/main/pages.js
@@ -146,15 +146,19 @@ const documentation = {
     isLinkOutside: true,
 };
 
-const getBasePages = (isCustomized) => [
-    overviewPage,
-    { ...networkSettings, pages: networkSettings.pages(isCustomized) },
-    administration,
-    isCustomized ? globalThreatStatistics : packageManagement,
-    advancedAdministration,
-    documentation,
-    aboutPage,
-];
+const getBasePages = (isCustomized) => {
+    const start = [
+        overviewPage,
+        { ...networkSettings, pages: networkSettings.pages(isCustomized) },
+        administration,
+    ];
+    const middle = isCustomized
+        ? [globalThreatStatistics]
+        : [packageManagement, advancedAdministration];
+    const end = [documentation, aboutPage];
+
+    return [...start, ...middle, ...end];
+};
 
 export default function getPages(isCustomized) {
     const PAGES = getBasePages(isCustomized);
-- 
2.52.0

