diff --git a/js/src/overview/Overview.js b/js/src/overview/Overview.js
index 6acefff7..6ef0655a 100644
--- a/js/src/overview/Overview.js
+++ b/js/src/overview/Overview.js
@@ -61,7 +61,7 @@ function OverviewCards({ packages, ws }) {
             <div className="row row-cols-1 row-cols-lg-3">
                 <AutomaticUpdatesCard />
                 <ThreatDetectionCard />
-                <DynamicFirewallCard enabled={displayCard(packages, "dynfw")} />
+                <DynamicFirewallCard enabled={displayCard(packages, "dynfw") || displayCard(packages, "dynfw_new")} />
             </div>
             <div className="row row-cols-1 row-cols-lg-3">
                 {displayCard(packages, "librespeed") &&
