From 74d387fcc98af5c634ab562949928de7dc220b20 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Sun, 11 Dec 2022 23:09:17 +0100
Subject: [PATCH 4/7] Use new path to get a serial number

---
 rescue.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rescue.sh b/rescue.sh
index db28c61..288fe72 100755
--- a/rescue.sh
+++ b/rescue.sh
@@ -108,6 +108,7 @@ mount -t devpts devpts /dev/pts
 mkdir -p /etc
 echo '/dev/mtd2 0 0x00010000' > /etc/fw_env.config
 SERIAL="$(cat /sys/bus/platform/devices/soc:internal-regs@d0000000:crypto@0/mox_serial_number)"
+[ -n "$SERIAL" ] || SERIAL="$(cat /sys/firmware/turris-mox-rwtm/serial_number)"
 MAC="$(cat /sys/class/net/eth0/address)"
 
 mkdir -p /root/.ssh
-- 
2.42.0

