--- a/meson.build
+++ b/meson.build
@@ -281,6 +281,11 @@ foreach libname, lib: { 'libknot': libkn
     l = ep / l.substring(p_len)
   endif
   conf_data.set_quoted(libname + '_SONAME', l / lib.get_variable(pkgconfig: 'soname'))
+  if rpath_opt == 'disabled'
+    conf_data.set_quoted(libname + '_SONAME', lib.get_variable(pkgconfig: 'soname'))
+  else
+    conf_data.set_quoted(libname + '_SONAME', l / lib.get_variable(pkgconfig: 'soname'))
+  endif
 endforeach
 
 kresconfig = configure_file(
