From d30f08f728c243f603383125854eac6b569d6d07 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Sat, 30 Aug 2025 22:25:48 +0200
Subject: [PATCH] omnia-ng: Force router workflow

---
 foris_controller_backends/web/__init__.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/foris_controller_backends/web/__init__.py b/foris_controller_backends/web/__init__.py
index e102863..cb59268 100644
--- a/foris_controller_backends/web/__init__.py
+++ b/foris_controller_backends/web/__init__.py
@@ -80,6 +80,8 @@ class WebUciCommands:
                 return profiles.Workflow.OLD.value
             elif SystemInfoFiles().get_contract() == "shield":
                 return profiles.Workflow.SHIELD.value
+            elif SystemInfoFiles().get_contract() == "omnia-ng":
+                return profiles.Workflow.ROUTER.value
         return profiles.Workflow.UNSET.value
 
     @staticmethod
@@ -117,6 +119,8 @@ class WebUciCommands:
                     return [profiles.Workflow.MIN.value, profiles.Workflow.BRIDGE.value]
         if model in ["turris", "omnia"]:
             return [profiles.Workflow.OLD.value]
+        if model in ["omnia-ng"]:
+            return [profiles.Workflow.ROUTER.value]
         return []
 
     @staticmethod
-- 
2.51.0

